lib/ci-helper.ts

```diff
@@ -626,15 +638,43 @@ export class CIHelper {
         }
     }
 
+    public async checkCommits(pr: IPullRequestInfo):
+        Promise<string> {
+        let comment: string = "";
+        const maxCommits = 30;
+        if (pr.commits && pr.commits > maxCommits) {
+            comment = `The pull request has ${pr.commits
+                } commits.  The max allowed is ${maxCommits
+                }.  Please squash some commits and 'git push --force'`;
+        }
+
+        return comment;
+    }
+
     public async handlePush(repositoryOwner: string, prNumber: number) {
         const pr = await this.github.getPRInfo(repositoryOwner, prNumber);
-        const gitGitGadget = await GitGitGadget.get(".");
-        if (!pr.hasComments && !gitGitGadget.isUserAllowed(pr.author)) {
-            const pullRequestURL =
-                `https://github.com/${repositoryOwner}/git/pull/${prNumber}`;
-            const welcome = (await readFile("res/WELCOME.md")).toString()
-                .replace(/\${username}/g, pr.author);
-            this.github.addPRComment(pullRequestURL, welcome);
+        const pullRequestURL = `https://github.com/${repositoryOwner
+                                }/git/pull/${prNumber}`;
+
+        try {
+            const gitGitGadget = await GitGitGadget.get(this.gggConfigDir,
+                                                        this.workDir);
+            if (!pr.hasComments && !gitGitGadget.isUserAllowed(pr.author)) {
+                const welcome = (await readFile("res/WELCOME.md")).toString()
+                        .replace(/\${username}/g, pr.author);
+                this.github.addPRComment(pullRequestURL, welcome);
+            }
+
+            const commitProblems = await this.checkCommits(pr);
+
+            if (commitProblems) {
+                this.github.addPRComment(pullRequestURL, commitProblems);
+                throw new Error(`Max commits exceeded.`);
+            }
+        } catch (e) {
+            const body = e.toString();
+            console.log(`Adding comment to ${pullRequestURL}:\n${body}`);
+            await this.github.addPRComment(pullRequestURL, body);
```

Comment:
Should we really throw an error here? It would appear to me as if this would have been enough:

```diff
+    if (commitProblems) {
+        this.github.addPRComment(pullRequestURL, commitProblems);
+    }
```

Reply:
This (the catch) is equivalent to the `addComment` function used in the big catch of  `handleComment`.  The change you noticed in your previous comment was the real reason for this change.

I agree throwing an error at 672 is not the best way to go since that is really intended for the job log and not as a PR comment.  This applies to all callers of checkCommits.

Reply:
try/catch removed - let misc-helper catch the errors.  If checkCommit returns false an error is thrown to make the CI check fail.  That will get the user's attention.